app/controllers/AddBusinessStartDateController.scala

```diff
@@ -0,0 +1,150 @@
+                                               implicit override val mcc: MessagesControllerComponents,
+                                               val ec: ExecutionContext)
+  extends ClientConfirmedController with I18nSupport with FeatureSwitching {
+
+  lazy val backUrl: String = routes.AddBusinessTradeController.show().url
+  lazy val backUrlAgent: String = routes.AddBusinessTradeController.showAgent().url
+
+  lazy val postAction: Call = routes.AddBusinessStartDateController.submit()
+  lazy val postActionAgent: Call = routes.AddBusinessStartDateController.submitAgent()
+
+  def show: Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit user =>
+      handleRequest(
+        isAgent = false,
+        backUrl = backUrl,
+        postAction = postAction,
+        itvcErrorHandler = itvcErrorHandler
+      )
+  }
+
+  def showAgent(): Action[AnyContent] = Authenticated.async {
+    implicit request =>
+      implicit user =>
+        getMtdItUserWithIncomeSources(incomeSourceDetailsService) flatMap {
+          implicit mtdItUser =>
+            handleRequest(
+              isAgent = true,
+              backUrl = backUrlAgent,
+              postAction = postActionAgent,
+              itvcErrorHandler = itvcErrorHandlerAgent
+            )
+        }
+  }
+
+  def handleRequest(isAgent: Boolean,
+                    backUrl: String,
+                    postAction: Call,
+                    itvcErrorHandler: ShowInternalServerError)
+                   (implicit user: MtdItUser[_]): Future[Result] = {
+    Future.successful(
+      if (isDisabled(IncomeSources)) {
+        Redirect(controllers.routes.HomeController.show())
+      } else {
+        Ok(addBusinessStartDate(
+          form = BusinessStartDateForm(),
+          postAction = postAction,
+          backUrl = backUrl,
+          isAgent = isAgent
+        ))
+      }
+    ) recover {
+      case ex: Exception =>
+        Logger("application").error(s"${if (isAgent) "[Agent]" else ""}" +
+          s"[AddBusinessStartDateController][handleRequest] - Error: ${ex.getMessage}")
+        itvcErrorHandler.showInternalServerError()
+    }
+  }
+
+  def submit: Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit request =>
+      Future.successful(
+        BusinessStartDateForm().bindFromRequest().fold(
+          formWithErrors =>
+            BadRequest(addBusinessStartDate(
+              form = formWithErrors,
+              postAction = postAction,
+              backUrl = backUrl,
+              isAgent = false
+            )),
+          formData =>
+            Redirect(routes.AddBusinessStartDateCheckController.show())
+              .addingToSession(SessionKeys.businessStartDate -> formData.date.toString)
```

Comment:
I don't see any immediate issues related to the format we store date, but just to note that this is a sensitive part